ChessBoardApp/ChessBoard/InputValidation.cs

```diff
@@ -0,0 +1,42 @@
+using Serilog;
+
+namespace ChessBoardApp.ChessBoard
+{
+
+
+    public class InputValidation
+    {
+        private readonly ILogger _logger;
+
+        public InputValidation(ILogger logger)
+        {
+            _logger = logger;
+        }
+        public class ValidationResult
```

Comment:
Question - it is a fairly good use case to have `ValidationResult` as nested class inside `InputValidation` class. What was your reason for this approach? What could be other use cases?

Reply:
Maybe nesting is better because we are not planning to have any other classes than `InputValidation` so it would not make sense to have `ValidationResult` class as a separate public class, since all of the instances would only be created inside `InputValidation` ? Also having the `ValidationResult` as a separate public class, is more error-prone, whereas now you can only use it with `InputValidation` instances. While writing this I googled, and I see it's called **encapsulation**.

Reply:
Good answer, resolved.